Source/Core/DolphinQt/Debugger/CodeViewWidget.cpp

```diff
@@ -189,25 +307,101 @@ void CodeViewWidget::Update()
+    setItem(i, CODE_VIEW_COLUMN_BREAKPOINT, bp_item);
+    setItem(i, CODE_VIEW_COLUMN_ADDRESS, addr_item);
+    setItem(i, CODE_VIEW_COLUMN_INSTRUCTION, ins_item);
+    setItem(i, CODE_VIEW_COLUMN_PARAMETERS, param_item);
+    setItem(i, CODE_VIEW_COLUMN_DESCRIPTION, description_item);
+    setItem(i, CODE_VIEW_COLUMN_BRANCH_ARROWS, branch_item);
 
     if (addr == GetAddress())
     {
       selectRow(addr_item->row());
     }
   }
 
+  CalculateBranchIndentation();
+
+  u32 max_indent = 0;
+  for (const CodeViewBranch& branch : m_branches)
+    max_indent = std::max(max_indent, branch.indentation);
+
   resizeColumnsToContents();
+  setColumnWidth(CODE_VIEW_COLUMN_BRANCH_ARROWS,
+                 static_cast<int>(WIDTH_PER_BRANCH_ARROW * (max_indent + 1)));
   g_symbolDB.FillInCallers();
 
   repaint();
   m_updating = false;
 }
 
+void CodeViewWidget::CalculateBranchIndentation()
+{
+  const size_t rows = rowCount();
+  const size_t columns = m_branches.size();
+  if (rows < 1 || columns < 1)
+    return;
+
+  // process in order of how much vertical space the drawn arrow would take up
+  // so shorter arrows go further to the left
+  const auto priority = [](const CodeViewBranch& b) {
+    return b.is_link ? 0 : (std::max(b.src_addr, b.dst_addr) - std::min(b.src_addr, b.dst_addr));
+  };
+  std::stable_sort(m_branches.begin(), m_branches.end(),
+                   [&priority](const CodeViewBranch& lhs, const CodeViewBranch& rhs) {
+                     return priority(lhs) < priority(rhs);
+                   });
+
+  // build a 2D lookup table representing the columns and rows the arrow could be drawn in
+  // and try to place all branch arrows in it as far left as possible
+  std::vector<bool> arrow_space_used(columns * rows, false);
+  const auto index = [&](u32 column, u32 row) { return column * rows + row; };
+  const u32 first_visible_addr = AddressForRow(0);
+  const u32 last_visible_addr = AddressForRow(static_cast<int>(rows - 1));
+  for (CodeViewBranch& branch : m_branches)
+  {
+    const u32 arrow_src_addr = branch.src_addr;
+    const u32 arrow_dst_addr = branch.is_link ? branch.src_addr : branch.dst_addr;
+    const u32 arrow_addr_lower = std::min(arrow_src_addr, arrow_dst_addr);
+    const u32 arrow_addr_higher = std::max(arrow_src_addr, arrow_dst_addr);
+    const bool is_visible =
+        last_visible_addr >= arrow_addr_lower || first_visible_addr <= arrow_addr_higher;
+    if (!is_visible)
+      continue;
+
+    const u32 arrow_first_visible_addr =
+        std::clamp(arrow_addr_lower, first_visible_addr, last_visible_addr);
+    const u32 arrow_last_visible_addr =
+        std::clamp(arrow_addr_higher, first_visible_addr, last_visible_addr);
+    const u32 arrow_first_visible_row = (arrow_first_visible_addr - first_visible_addr) / 4;
+    const u32 arrow_last_visible_row = (arrow_last_visible_addr - first_visible_addr) / 4;
+
+    const std::optional<u32> free_column = [&]() -> std::optional<u32> {
```

Comment:
```suggestion
    const std::optional<u32> free_column = [&] -> std::optional<u32> {
```

(and maybe use auto here, since the return type is already explicitly written out?)

Reply:
The parens here seem to be mandatory when explicitly defining the return type. I'll use the `auto` though.